### DISCUSSION TITLE
【K2P未修改】为什么使用Lede编译的固件和immortalWrt编译的闭源固件差别这么大？

### DISCUSSION THREAD

**Discussion by asiaon123** [immortalwrt/immortalwrt#528] [Q&A]
- Lede：
    无线可以达到500M以上
    可以使用uci wifi配置
- ImmortalWrt
    无线极其不稳定，最多能到150M
    不能使用uci wifi配置

**Comment by 1715173329** [comment]
wwwwait... which branch are you using here?

**Reply by 1715173329** [reply]
btw, mt_wifi is crappy, if we add uci support for it, then much more ugly hacks will come up.
For now, the newer mt76 driver is good enough.

**Reply by asiaon123** [reply]
> wwwwait... which branch are you using here?

I have tried all the branches inclucing master, 21.02 and 18.06, but they come to the same end.

**Reply by asiaon123** [reply]
> btw, mt_wifi is crappy, if we add uci support for it, then much more ugly hacks will come up. For now, the newer mt76 driver is good enough.

I also tried mt76 drive, but it looks like it's not working well comparing with the mt_wifi.

**Reply by asiaon123** [reply]
可以实锤，你的这个优化可能还是有点问题的，根据我自己手里的机器测试，3台机器都远不如lede的固件
- hc5661 开源驱动
    信号质量差
- hc5962 开源驱动
    无线质量差，易断线
    断电起不开，需要多次启动才有概率启动开
- k2p 闭源驱动
    无线质量差，不稳定

这里的编译都基于2102

lede的话除了界面差一点，有的软件版本低了一点，基本没啥问题，关键无线信号很好

不知道这边的固件什么原因，做了什么负优化还是什么其他的影响到了

**Reply by asiaon123** [reply]
抱歉我在这儿猛夸lede，但是你这个固件确实问题挺大的

**Reply by chobitsos** [reply]
lean的源码比这里的坑更多，我已经放弃lean大神的源码了，投入到immortalwrt这边来了，这边@1715173329 大神回答问题比较快
坑也比较少，而且完全适配了lean大神的packges,深度开发的话还是immortalwrt比较适合一点
两边各有优缺点，无线问题那边好用就去那边的

**Comment by 1715173329** [comment]
padavan 是 ramips 最后的归宿。

**Reply by chobitsos** [reply]
padavan 我放弃了，因为没有X86，太难了，而且坑也不少，用padavan 就像用官方的源码和凉白开一样

**Reply by thisdk** [reply]
赞同,我的K2P已经用hanwckf的padavan稳定做AP半年了.基本不重启.因为cpu还是太弱了.没啥好折腾,想折腾还是要换arm/x86
